**Задача 1.** 
*Даны две независимые выборки. Не соблюдается условие нормальности
x1 380,420,290
y1 140,360,200,900
Сделайте вывод по результатам, полученным с помощью функции, имеются ли статистические различия между группами?*

Формулируем гипотезу $Н_0$ как "статистичесих различий нет". Так как у нас 2 независимые выборки проверим гипотезу с помощью критерия Манна – Уитни, $\alpha=0.05$

In [1]:
import numpy as np
import scipy.stats as stats

In [2]:
x=np.array([380,420,290])
y=np.array([140,360,200,900])
stats.mannwhitneyu(x, y)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

**Ответ**

$pvalue>\alpha(0.05)$
следовательно статистически значимых различий нет, гипотезу
$Н_0$  не отвергаем

---

**Задача 2.** 
*сследовалось влияние препарата на уровень давления пациентов. Сначала
измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть
ли статистически значимые различия между измерениями давления? В выборках не соблюдается условие нормальности*

*1е измерение до приема препарата: 150, 160, 165, 145, 155

*2е измерение через 10 минут: 140, 155, 150, 130, 135

*3е измерение через 30 минут: 130, 130, 120, 130, 125*          

Формулируем гипотезу $Н_0$ как "статистичесих различий нет, препарат не влияет на давление". Так как у нас 3е повторных измерений одной группы (зависимые выборки) проверим гипотезу с помощью критерия Фридмана, $\alpha=0.05$


In [3]:
x1=np.array([150, 160, 165, 145, 155])
x2=np.array([140, 155, 150, 130, 135])
x3=np.array([130, 130, 120, 130, 125])

stats.friedmanchisquare(x1, x2, x3)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

$pvalue<\alpha(0.05)$
статистически значимых различия есть
$Н_0$ отвергаем, препарат влияет на давление

**Задача 3.** 
*Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.*

Выборка стала парной - используем критерий Уилкоксона


In [4]:
x1=np.array([150, 160, 165, 145, 155])
x2=np.array([140, 155, 150, 130, 135])

stats.wilcoxon(x1, x2)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

**Ответ:**

Так как при таком измерении $pvalue>\alpha(0.05)$ - статистически значимых различий нет, $Н_0$ не отвергаем, препарат не влияет на давление

---

**Задача 4.** 
*Даны 3 группы учеников плавания. Не соблюдается условие нормальности.
В 1 группе время на дистанцию 50 м составляют:
56, 60, 62, 55, 71, 67, 59, 58, 64, 67
Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53
Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54
Есть ли статистически значимые различия между группами?*

Формулируем гипотезу $Н_0$ как "статистичесих различий нет". Так как у нас 3 независимые выборки проверим гипотезу с помощью критерия Крускала –Уоллиса, $\alpha=0.05$

In [8]:
x1=np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
x2=np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
x3=np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])

stats.kruskal(x1,x2,x3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

**Ответ:**

так как $pvalue>\alpha(0.05)$ - статистически значимых различий нет, 
$Н_0$ не отвергаем

---

**Задача 5.** 
*Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить
данную гипотезу, если известно, что размеры изделий подчинены нормальному закону
распределения. Объем выборки 10, уровень статистической значимости 5%
2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34*

Соствим гипотезы

$Н_0$ :  среднее арифметическое партии равно 2,5 см или  $\mu= 2,5$

$Н_1$ :  $\mu \neq 2.5$

расчитаем наблюдаемое значение tn по формуле

$$t = \dfrac{\overline{X} - a}{\sigma_X / \sqrt{n}}$$

In [6]:
s = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])

n = 10

mean = s.mean()
std = s.std(ddof=1)

tn = (mean-2.5)/(std/np.sqrt(n))
tn

0.5630613661802959

Доверительная вероятность $= 0.95$, число степеней свободы = $n - 1 = 9$
находим критерий Стьюдента табличный tt: 

In [7]:
tt=stats.t.ppf(0.95,9)
tt

1.8331129326536335

**Ответ**

Так как табличное значение больше наблюдаемого tt > tn ( 1.833 > 0.56), то гипотеза верна и партия изготавливается со средним арифметическим 2,5 см..